glsk/glsk-document-cse/src/main/java/com/powsybl/glsk/cse/CseGlskDocument.java

```diff
@@ -9,70 +9,154 @@
+        Map<String, List<GlskPoint>> cseGlskPointsExport = getGlskPointsFromTimeSeries(nativeGskDocument.getTimeSeriesExport());
+
+        if (!useCalculationDirections || calculationDirectionsAbsent(nativeGskDocument)) {
+            // "default" mode : all countries are considered in full import (use TimeSeries for all)
+            this.cseGlskPoints.putAll(cseGlskPointsStandard);
+        } else {
+            // Extract CalculationDirections
+            List<CalculationDirectionType> calculationDirections = nativeGskDocument.getCalculationDirections().get(0).getCalculationDirection();
+            Map<String, List<String>> countriesInAndOutArea = getCountriesInAndOutArea(calculationDirections);
+
+            // Use data from cseGlskPointsStandard or cseGlskPointsExport depending on CalculationDirections
+            fillGlskPointsForExportCorner(cseGlskPointsStandard, cseGlskPointsExport, countriesInAndOutArea);
+        }
+    }
+
+    private static Map<String, List<GlskPoint>> getGlskPointsFromTimeSeries(List<TimeSeriesType> timeSeries) {
+        Map<String, List<GlskPoint>> cseGlskPoints = new TreeMap<>();
+        if (timeSeries == null) {
+            return cseGlskPoints;
+        }
+
+        timeSeries.stream()
+            .map(CseGlskPoint::new)
+            .forEach(glskPoint -> {
                 cseGlskPoints.computeIfAbsent(glskPoint.getSubjectDomainmRID(), area -> new ArrayList<>());
                 cseGlskPoints.get(glskPoint.getSubjectDomainmRID()).add(glskPoint);
-            }
-        }
+            });
+        return cseGlskPoints;
+    }
+
+    private static boolean calculationDirectionsAbsent(GSKDocument nativeGskDocument) {
+        return nativeGskDocument.getCalculationDirections() == null
+            || nativeGskDocument.getCalculationDirections().isEmpty()
+            || nativeGskDocument.getCalculationDirections().get(0).getCalculationDirection() == null
+            || nativeGskDocument.getCalculationDirections().get(0).getCalculationDirection().isEmpty();
+    }
+
+    private static Map<String, List<String>> getCountriesInAndOutArea(List<CalculationDirectionType> calculationDirections) {
+        String italyEIC = new CountryEICode(Country.IT).getCode();
+        List<String> countriesInArea = new ArrayList<>();
+        List<String> countriesOutArea = new ArrayList<>();
+
+        calculationDirections.stream()
+            .map(cd -> cd.getInArea().getV())
+            .filter(countryEIC -> !countryEIC.equals(italyEIC))
+            .forEach(countriesInArea::add);
+
+        countriesInArea.add(italyEIC);
+
+        calculationDirections.stream()
+            .map(cd -> cd.getOutArea().getV())
+            .filter(countryEIC -> !countryEIC.equals(italyEIC))
+            .forEach(countriesOutArea::add);
+
+        return Map.of(COUNTRIES_IN_AREA_KEY, countriesInArea,
+            COUNTRIES_OUT_AREA_KEY, countriesOutArea);
```

Comment:
For Export Corner developments, the CalculationDirections (representing which countries have activated Export Corner excahnges or not) should contain only directions from a country to Italy or vice-versa. E.g.: {InArea: France, OutArea: Italy} if France has activated Export Corner exchanges, or {InArea: Italy, OutArea: France} if France has not activated Export Corner exchanges.
There is no sense to put {InArea: Italy, OutArea: Italy} because no exchange is made from Italy to itself.
But we need to handle Italy because we need to retrieve its TimeSeries. As the rules for Italy (concerning TimeSeries retrieval) are the same as the rules for countries having activated Export Corner exchanges, I decided to put it in the `countriesInArea` List.